phd-tests/framework/src/test_vm/mod.rs

```diff
@@ -142,6 +142,51 @@ enum VmState {
     Ensured { serial: SerialConsole },
 }
 
+/// Both the output and status of a command.
+pub struct ShellOutput {
+    status: u16,
+    output: String,
+}
+
+impl ShellOutput {
+    /// Consume this [`ShellOutput`], returning the command's output as text
+    /// if the command completed successfully, or an error if it did not.
+    pub fn expect_ok(self) -> Result<String> {
+        if self.status == 0 {
+            Ok(self.output)
+        } else {
+            Err(anyhow!("command exited with non-zero status: {}", self.status))
+        }
+    }
+
+    /// Consume this [`ShellOutput`], returning the command's output as text
+    /// if the command exited with non-zero status, or an error if it exited
+    /// with status zero.
+    pub fn expect_err(self) -> Result<String> {
+        if self.status != 0 {
+            Ok(self.output)
+        } else {
+            Err(anyhow!("command unexpectedly succeeded"))
+        }
+    }
+
+    pub fn ignore_status(self) -> String {
```

Comment:
I'm on the fence about this name; I do think it's nice that it makes it clear that you're ignoring the status, but I think it might be better named `into_output` or something, because "ignore_status" makes it clear what you're ignoring, but not what you're _getting_, which I think is useful to make clear at the callsite...

Also, I can write something like this:
```rust
let output: ShellOutput = ...;
if output.status > 0 {
   // ... 
}
do_something_with_output(output.ignore_status())
```
and i'm not *really* "ignoring" the status at all, i'm just handling it separately.

YES I KNOW THIS IS INCREDIBLY PEDANTIC OF ME AND IT DOESN'T ACTUALLY MATTER

Reply:
there's still an `ignore_result` but it's configuring the command execution Future now where i think the relationship is a lot less ambiguous. gonna resolve this but if the functions on `ShellOutputExecutor` still seem a bit weird please check me.